engine/src/main/java/org/hibernate/search/engine/service/impl/StandardServiceManager.java

```diff
@@ -44,6 +44,7 @@
 	private final Map<Class<? extends Service>, String> defaultServices;
 
 	private volatile boolean allServicesReleased = false;
+	private volatile ClassLoaderService classloaderService = null;
```

Comment:
ok, that should be doable. Remember a volatile read is not an issue on x86 though